src/nvim/ex_getln.c

```diff
@@ -3587,8 +3607,10 @@ void redrawcmd(void)
   // Don't use more prompt, truncate the cmdline if it doesn't fit.
   msg_no_more = TRUE;
   draw_cmdline(0, ccline.cmdlen);
-  msg_clr_eos();
-  msg_no_more = FALSE;
+  if (p_ch > 0) {
+    msg_clr_eos();
+  }
```

Comment:
Instead of adding `p_ch > 0` checks before calling `msg_clr_eos` in multiple places, why not add the check inside `msg_clr_eos`?